superset/views/datasource/views.py

```diff
@@ -228,7 +228,10 @@ class DatasetEditor(BaseSupersetView):
     @has_access
     @permission_name("read")
     def root(self) -> FlaskResponse:
-        return super().render_app_template()
+        dev = request.args.get("testing")
+        if dev is not None:
+            return super().render_app_template()
+        return redirect("/")
```

Comment:
what is this? ^^^

Reply:
This is from the revert, not my change

Reply:
I checked blame, the original code is from this PR: #20869